File: C++/59-按之字形顺序打印二叉树.cpp
/*
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
    TreeNode(int x) :
            val(x), left(NULL), right(NULL) {
    }
};
*/
class Solution
{
public:
    vector<vector<int>> Print(TreeNode *pRoot)
    {
        vector<vector<int>> ret;
        queue<TreeNode *> q;
        q.push(pRoot);
        if (!pRoot)
            return ret;
        int l2r = 1;
        while (!q.empty())
        {
            vector<int> t_ret;
            queue<TreeNode *> t_q;
            while (!q.empty())
            {
                TreeNode *node = q.front();
                q.pop();
                if (l2r)
                    t_ret.push_back(node->val);
                else
                    t_ret.insert(t_ret.begin(), node->val);
                if (node->left)
                    t_q.push(node->left);
                if (node->right)
                    t_q.push(node->right);
            }
            q = t_q;
            ret.push_back(t_ret);
            l2r = 1 - l2r;
        }
        return ret;
    }
};

File: Python/59-按之字形顺序打印二叉树.py
# -*- coding:utf-8 -*-
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class Solution:
    def Print(self, pRoot):
        # write code here
        if not pRoot:
            return []
        queue = [pRoot]
        l2r = 1
        ret = []
        while queue:
            t_ret = []
            t_queue = []
            for _ in range(len(queue)):
                node = queue.pop(0)
                if l2r:
                    t_ret.append(node.val)
                else:
                    t_ret.insert(0, node.val)
                if node.left:
                    t_queue.append(node.left)
                if node.right:
                    t_queue.append(node.right)
            queue = t_queue
            ret.append(t_ret)
            l2r = 1 - l2r
        return ret